api/v1alpha1/accesslogging_types.go

```diff
@@ -30,8 +30,30 @@ type ProxyAccessLogSetting struct {
 	// +kubebuilder:validation:MinItems=1
 	// +kubebuilder:validation:MaxItems=50
 	Sinks []ProxyAccessLogSink `json:"sinks"`
+	// Type defines the component emitting the accesslog, such as Listener and Route.
+	// If not defined, the setting would apply to all accesslog types supported by envoy gateway
+	// +kubebuilder:validation:Enum=Listener
+	// +optional
+	// +notImplementedHide
+	Type *ProxyAccessLogType `json:"type,omitempty"`
```

Comment:
also if we take the `type` route and both route and listener types exist, we'll have to define the `sinks` twice ?

Reply:
clarified the behavior in the doc string and PR desc. When users select a type, the definition will replace any EG defaults for that component and apply only to that component. If type is not specified - behavior would be the same as today (custom format applied to both routes and listeners, default listener format will be configured as well).